### PR TITLE
[SPARK-7444][Tests] Eliminate noisy css warn/error logs for UISeleniumSuite

### PR DESCRIPTION
**Pull request by zsxwing**
Eliminate the following noisy logs for `UISeleniumSuite`:

```
15/05/07 10:09:50.196 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS error: 'http://192.168.0.170:4040/static/bootstrap.min.css' [793:167] Error in style rule. (Invalid token "*". Was expecting one of: <EOF>, <S>, <IDENT>, "}", ";".)
15/05/07 10:09:50.196 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS warning: 'http://192.168.0.170:4040/static/bootstrap.min.css' [793:167] Ignoring the following declarations in this rule.
15/05/07 10:09:50.197 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS error: 'http://192.168.0.170:4040/static/bootstrap.min.css' [799:325] Error in style rule. (Invalid token "*". Was expecting one of: <EOF>, <S>, <IDENT>, "}", ";".)
15/05/07 10:09:50.197 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS warning: 'http://192.168.0.170:4040/static/bootstrap.min.css' [799:325] Ignoring the following declarations in this rule.
15/05/07 10:09:50.198 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS error: 'http://192.168.0.170:4040/static/bootstrap.min.css' [805:18] Error in style rule. (Invalid token "*". Was expecting one of: <EOF>, <S>, <IDENT>, "}", ";".)
15/05/07 10:09:50.198 pool-1-thread-1-ScalaTest-running-UISeleniumSuite WARN DefaultCssErrorHandler: CSS warning: 'http://192.168.0.170:4040/static/bootstrap.min.css' [805:18] Ignoring the following declarations in this rule.
```

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tianyi @marmbrus @JoshRosen

File: core/src/main/resources/org/apache/spark/ui/static/streaming-page.css [hunk 1]
```diff
@@ -31,7 +31,7 @@
 }
 
 .tooltip-inner {
-  max-width: 500px !important; // Make sure we only have one line tooltip
```
Comment (zsxwing) [inline core/src/main/resources/org/apache/spark/ui/static/streaming-page.css]: I just found this incorrect comment as per the error log. So I think we should not totally turn off all logs. That's why this PR only ignores logs from "bootstrap.min.css" and "vis.min.css".

**Comment by tdas** [comment]
Can you elaborate a little on what the `DefaultCssErrorHandler` solves.

File: core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala [hunk 2]
```diff
@@ -44,7 +46,33 @@ class UISeleniumSuite extends FunSuite with WebBrowser with Matchers with Before
   implicit var webDriver: WebDriver = _
 
   override def beforeAll(): Unit = {
-    webDriver = new HtmlUnitDriver
+    webDriver = new HtmlUnitDriver {
+
+      getWebClient.setCssErrorHandler(new DefaultCssErrorHandler {
```
Comment (tdas) [inline core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala]: This new handler's code is triplicated. Cant you just define a class?

Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala]: Is there a place for common test classes for all projects?

**Comment by zsxwing** [comment]
@tdas DefaultCssErrorHandler outputs all css warn/error logs. This PR just changes its behavior to: if  the warn/error is from "bootstrap.min.css" and "vis.min.css", which should be a false alarm, just ignore it.

**Comment by tianyi** [comment]
@zsxwing Maybe we could define a log level instead of whitelist?
because:
1 We should not ignore all kind of logs including error and fatal error.
2 It will be hard for us to maintain a whitelist of all CSS filenames.

**Comment by zsxwing** [comment]
@tianyi here I just ignored the trusted libraries's filename, "bootstrap.min.css" and "vis.min.css".  I'm also against ignoring all CSS filenames.

> Maybe we could define a log level instead of whitelist?

Which level are you suggesting? They are "WARN". I cannot find a better level for it.

File: core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala (hunk 2 above)
Comment (JoshRosen) [inline core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala]: We can put the test class into core then have the other projects declare a test-jar test dependency on core.  I think that SQL uses this pattern to share test code between SQL core and catalyst.

Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala]: Does Core build a test-jar now?

**Comment by marmbrus** [comment]
Thanks, merging to master!

**Pull request closed without merging**